### PR TITLE
Implement rpc for Service related stuff

### PR DESCRIPTION
**Pull request by resouer**
```
    // ServiceList  implements   /service/list
    rpc ServiceList(ServiceListRequest) returns (ServiceListResponse) {}
    // ServiceAdd implements  POST   /service/add
    rpc ServiceAdd(ServiceAddRequest) returns (ServiceAddResponse){}
    // ServiceDelete implements  DELETE /service
    rpc ServiceDelete(ServiceDelRequest) returns (ServiceDelResponse){}
    // ServiceUpdate updates an existing service
    rpc ServiceUpdate(ServiceUpdateRequest) returns (ServiceUpdateResponse){}
```

### CODE REVIEW

File: types/types.proto [hunk 1]
```diff
@@ -498,6 +498,38 @@ message ImageRemoveResponse {
...
+  repeated UserService services = 1;
+}
+
+message ServiceListRequest {
+  string podID =1;
+}
+
+message ServiceAddRequest {
+  string podID =1;
+  string data  =2;
```
Comment (feiskyer) [inline types/types.proto]: Is it possible to replace `data` with `UserService`?

File: types/types.proto [hunk 2]
```diff
@@ -311,14 +311,14 @@ message UserInterface {
 }
 
 message UserServiceBackend {
-  string hostip   = 1;
-  int32 hostport  = 2;
+  string hostIP   = 1;
+  int32 hostPort  = 2;
```
Comment (feiskyer) [inline types/types.proto]: nit: modifying those fields will also change restful apis.

File: types/types.proto (hunk 1 above)
Comment (resouer) [inline types/types.proto]: Sure, will add new commits to fix this

File: types/types.proto [hunk 3]
```diff
@@ -567,11 +599,15 @@ service PublicAPI {
     // Wait gets the exit code of the specified container
     rpc Wait(WaitRequest) returns (WaitResponse) {}
     // TODO: TTYResize resizes the tty of the specified container
-
-    // TODO: ServiceList gets a list of services
-    // TODO: ServiceCreate creates a new service
-    // TODO: ServiceUpdate updates an existing service
-    // TODO: ServiceDelete deletes the specified service
+    
+    // ServiceList  implements   /service/list
```
Comment (feiskyer) [inline types/types.proto]: Please update those docs as `ServiceList gets a list of services`

File: types/types.proto [hunk 4]
```diff
@@ -515,6 +515,38 @@ message VersionResponse {
...
+  repeated UserService services = 1;
+}
+
+message ServiceListRequest {
+  string podID =1;
+}
+
+message ServiceAddRequest {
+  string podID                 =1;
+  repeated UserService services  =2;
```
Comment (feiskyer) [inline types/types.proto]: Please indent those field by `=`

**Comment by feiskyer** [comment]
LGTM. Thanks.

**Pull request merged**